File: backend/entity/contract.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

// entity/contract.go
type Contract struct {
	gorm.Model
	Start_Date time.Time `json:"start_date"`
	End_Date   time.Time `json:"end_date"`
	Rate       float64   `json:"rate" gorm:"type:decimal(10,2)"`
	RenewalPending   bool       `json:"renewal_pending" gorm:"default:false"`
	RenewalMonths    *int       `json:"renewal_months"`
	RenewalStartDate *time.Time `json:"renewal_start_date"`
	RenewalEndDate   *time.Time `json:"renewal_end_date"`
	RenewalRate      *float64   `json:"renewal_rate"`
	RenewalStatus    *string    `json:"renewal_status"` // "pending" | "approved" | "rejected"

	RoomID *uint `json:"Room_ID"`
	Room    Room    // ความสัมพันธ์ย้อนกลับ

	Admin_ID *uint
	Admin    Admin `gorm:"foreignKey:Admin_ID"`

	StudentID *uint
	Student   *Student `gorm:"foreignKey:StudentID"`
	
	PaymentID *uint
	Payment   *Payment `gorm:"foreignKey:PaymentID"`


}

File: frontend/src/interfaces/Contract.ts
import type { StudentInterface } from "./Student";

export interface ContractInterface {
  ID: number;
  start_date: string;
  end_date: string;
  rate?: number;
  Student_ID?: number;
  Student?: StudentInterface;

  renewal_pending?: boolean;
  renewal_status?: "pending" | "approved" | "rejected" | null;
  renewal_start_date?: string | null;
  renewal_end_date?: string | null;
  renewal_months?: number | null;
  renewal_rate?: number | null;
}

type ContractWithRenewal = ContractInterface & {
  // existing renewal fields...
  renewal_pending?: boolean;
  renewal_status?: string | null;
  renewal_start_date?: string | null;
  renewal_end_date?: string | null;
  renewal_months?: number | null;
  renewal_rate?: number | null;

  // 🔽 add termination fields
  termination_pending?: boolean | null;
  termination_status?: string | null;              // "pending" | "approved" | "rejected"
  termination_effective_date?: string | null;
  termination_reason?: string | null;
  termination_penalty?: number | null;
  termination_refund?: number | null;
};
